source/beamline/SGM/SGMHexapodTransformedAxis.cpp

```diff
@@ -29,12 +32,196 @@ SGMHexapodTransformedAxis::SGMHexapodTransformedAxis(AxisDesignation axis,
+		endMoveActions->addSubAction(AMActionSupport::buildControlMoveAction(globalZAxis_, newGlobalEndSetpoints.z()));
+
+
+		action->addSubAction(endMoveActions);
+
+		// Wait for the parameters to take
+		AMListAction3* endWaitActions = new AMListAction3(new AMListActionInfo3("Waiting for parameter controls for end",
+		                                                                     "Waiting for parameter controls for end"),
+		                                               AMListAction3::Parallel);
+
+		endWaitActions->addSubAction(AMActionSupport::buildControlWaitAction(globalXAxis_, newGlobalEndSetpoints.x(), 2, AMControlWaitActionInfo::MatchWithinTolerance));
+		endWaitActions->addSubAction(AMActionSupport::buildControlWaitAction(globalYAxis_, newGlobalEndSetpoints.y(), 2, AMControlWaitActionInfo::MatchWithinTolerance));
+		endWaitActions->addSubAction(AMActionSupport::buildControlWaitAction(globalZAxis_, newGlobalEndSetpoints.z(), 2, AMControlWaitActionInfo::MatchWithinTolerance));
+
+		action->addSubAction(endWaitActions);
+		// Save the previous velocity before we alter it
+		lastSavedVelocity_ = systemVelocity_->value();
+
+		// Setting the system velocity
+		double velocity = qAbs(endPoint - startPoint) / deltaTime;
+		action->addSubAction(AMActionSupport::buildControlMoveAction(systemVelocity_, velocity));
+		action->addSubAction(AMActionSupport::buildControlWaitAction(systemVelocity_, velocity, 2, AMControlWaitActionInfo::MatchWithinTolerance));
+
+		// Setting up the data recorder
+		double requiredRate = HEXAPOD_RECORDER_POINTS_PER_MOVE / deltaTime;
+		action->addSubAction(AMActionSupport::buildControlMoveAction(dataRecorderRate_, requiredRate));
+		action->addSubAction(AMActionSupport::buildControlWaitAction(dataRecorderRate_, requiredRate, 2.0, AMControlWaitActionInfo::MatchWithinTolerance));
+
+		action->addSubAction(AMActionSupport::buildControlMoveAction(dataRecorderStatus_, 1));
+		action->addSubAction(AMActionSupport::buildControlWaitAction(dataRecorderStatus_, 1, 2.0, AMControlWaitActionInfo::MatchWithinTolerance));
+
+	}
+
+	return action;
+}
+
+AMAction3 * SGMHexapodTransformedAxis::createInitializeCoordinatedMovementActions()
+{
+	return 0;
+}
+
+AMAction3 * SGMHexapodTransformedAxis::createStartCoordinatedMovementActions()
+{
+	return  AMActionSupport::buildControlMoveAction(trajectoryStartControl_, 1);
+}
+
+AMAction3 * SGMHexapodTransformedAxis::createWaitForCompletionActions()
+{
+	AMListAction3* waitActions = new AMListAction3(new AMListActionInfo3("Waiting for coordinated move",
+	                                                                     "Waiting for coordinated move"),
+	                                               AMListAction3::Sequential);
+
+	// We need to add 1.5 seconds to the delta time, as the data recorder is slow
+	AMAction3* positionWaitAction = AMActionSupport::buildControlWaitAction(this, lastEndPoint_, lastDeltaTime_+1.5, AMControlWaitActionInfo::MatchWithinTolerance);
```

Comment:
+1.5
With the data recorder taking about a second to do its retrieval whatever the length of time over which the motion was performed, I opted for a flat amount.